client/Client.py

```diff
@@ -39,6 +43,42 @@ def join_queue(self):
             self.socket.shutdown(socket.SHUT_RDWR)
```

Comment:
Why does socket shutdown straight after connect?

Reply:
Right, there shouldn't be a shutdown there. It was meant to shut down if any parsing ValueError occured, but server has changed way of communication, so the entire try/catch and shutdown unnecessary.